app/scripts/controllers/network/network-controller.test.ts

```diff
@@ -8239,18 +8063,26 @@ function mockCreateNetworkClient() {
 }
 
 /**
- * Builds the set of controller messengers that recognizes the events that
- * NetworkController emits: one designed to be used directly by
- * NetworkController, and one designed to be used in tests.
+ * Builds a controller messenger.
  *
  * @returns The controller messenger.
  */
-function buildMessengerGroup() {
-  const unrestrictedMessenger = new ControllerMessenger<
-    never,
-    NetworkControllerEvent
-  >();
-  const restrictedMessenger = unrestrictedMessenger.getRestricted<
+function buildMessenger() {
+  return new ControllerMessenger<never, NetworkControllerEvent>();
+}
+
+/**
+ * Despite the signature of its constructor, NetworkController must take an
+ * Infura project ID. The object that this function returns is mixed into the
+ * options first when a NetworkController is instantiated in tests.
+ *
+ * @param messenger - A controller messenger.
+ * @returns The controller options.
+ */
+function buildDefaultNetworkControllerOptions(
```

Comment:
Hmm. I'm realizing now that the tests are a bit out of sync with respect to how the `withController` helper is written exactly. I like the way that we have it now in the `core` repo as it's simpler, so maybe we make it more consistent from this side? In other words what if `withController` were:

``` typescript
async function withController<ReturnValue>(
  ...args: WithControllerArgs<ReturnValue>
): Promise<ReturnValue> {
  const [{ ...rest }, fn] = args.length === 2 ? args : [{}, args[0]];
  const messenger = buildMessenger();
  const restrictedMessenger = buildNetworkControllerMessenger(messenger);
  const controller = new NetworkController({
    messenger: restrictedMessenger,
    trackMetaMetricsEvent: jest.fn(),
    infuraProjectId: 'infura-project-id',
    ...rest,
  });
  try {
    return await fn({ controller, messenger });
  } finally {
    await controller.destroy();
  }
}
```

Then we wouldn't need `buildDefaultNetworkControllerOptions` or `DEFAULT_INFURA_PROJECT_ID`. But in its place, we'd introduce `buildNetworkControllerMessenger`.

Reply:
Good idea! I had noticed this difference, but thought it would be more work to resolve than it ended up being.

Done in https://github.com/MetaMask/metamask-extension/pull/18901/commits/3e464e7daa1af69770c1dc740e0e3d815fc6492b